tests/test_TFUtil.py

```diff
@@ -469,6 +469,24 @@ def test_Data_copy_compatible_to_batch_feature_is_dynamic():
   assert_equal(t.get_size_dim_tag(0), dec.get_time_dim_tag())
 
 
+def test_Data_get_common_data_modified_time_dim():
+  d1 = Data(name='t', shape=(None, 32, 128), dtype='float32', auto_create_placeholders=True)
+  d2 = Data(name='r', shape=(None, 32, 128), dtype='float32', auto_create_placeholders=True)
+  d1.size_placeholder[0] -= 1
+  d2.size_placeholder[0] -= 1
+
+  from TFUtil import DimensionTag
+  tag1 = DimensionTag(description="slice1",
+                      kind=DimensionTag.Types.Spatial)
+  tag1.set_tag_on_size_tensor(d1.size_placeholder[0])
+  tag2 = DimensionTag(description="slice2",
+                      kind=DimensionTag.Types.Spatial)
+  tag2.set_tag_on_size_tensor(d2.size_placeholder[0])
+  common = Data.get_common_data([d1, d2], warnings_out=sys.stdout)
+  assert d1.shape == d2.shape
+  assert common.shape == d1.shape
```

Comment:
This assert is wrong. They have different time dims, and thus the `common.shape` should be `(None, None, 32, 128)`.

Reply:
But that's my case, I have d1 and d2, with different time dims. And I want to combine them in the CombineLayer. And I don't want  `common.shape` have an extra dim. You see what I mean?

Reply:
That does not make sense. If d1 is e.g. of shape (3,T1=4,5), and d2 is of shape (3,T2=7,5), you cannot combine T1 and T2. It has to become (3,T1,T2,5).

Reply:
But I take care of that. I don't need that check. Do you see more high level tests?

Reply:
My topology guarantees that T1 and T2 are equal despite they have different tags

Reply:
But then this is a bug/problem with your setup, or elsewhere. They should have the same tag if they are the same. If they have different tags, that means they are different. That's the whole point of having this.

Reply:
I added a test with the attention layer `test_self_att_config`. What's wrong with it? Indeed the input is split into key, querry, and value. Each is processed differently and your new tagging system assigns them different tags. And that makes impossible to combine them later.
It worked fine with old version of returnn, so there's no bugs in the setups.

Reply:
As I see the problem, the tagging doesn't work properly with all layers, which modify the data shape. E.g. window layer and slice layer. Maybe some others.

Reply:
This thread here is about `test_Data_get_common_data_modified_time_dim`, nothing else. I'm explaining you now multiple times that this test is invalid, and the behavior is correct, and as intended.
For the other tests, or in general, I already asked you, or told you, that you should show more specifically, where you think it goes wrong. Where exactly. What is the input there, what is the output. Exactly at what step is it wrong?

Reply:
> This thread here is about `test_Data_get_common_data_modified_time_dim`, nothing else. I'm explaining you now multiple times that this test is invalid, and the behavior is correct, and as intended.

Ok